// cypress-mongodb/dist/commands/find.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.findMany = exports.findOne = void 0;
function findOne(query, collection, database) {
    var args = {
        uri: Cypress.env('MONGODB_URI'),
        database: Cypress.env('MONGODB_DATABASE'),
        collection: Cypress.env('MONGODB_COLLECTION'),
        pipeline: [],
    };
    if (!query) {
        throw new Error('Query must be specified');
    }
    if (collection) {
        args.collection = collection;
    }
    else if (!args.collection) {
        throw new Error('Collection not specified');
    }
    if (database) {
        args.database = database;
    }
    else if (!args.database) {
        throw new Error('Database not specified');
    }
    return cy
        .task('findOne', {
        uri: args.uri,
        pipeline: query,
        collection: args.collection,
        database: args.database,
    })
        .then(function (result) {
        return result;
    });
}
exports.findOne = findOne;
function findMany(query, collection, database) {
    var args = {
        uri: Cypress.env('MONGODB_URI'),
        database: Cypress.env('MONGODB_DATABASE'),
        collection: Cypress.env('MONGODB_COLLECTION'),
        pipeline: [],
    };
    if (!query) {
        throw new Error('Query must be specified');
    }
    if (collection) {
        args.collection = collection;
    }
    else if (!args.collection) {
        throw new Error('Collection not specified');
    }
    if (database) {
        args.database = database;
    }
    else if (!args.database) {
        throw new Error('Database not specified');
    }
    return cy
        .task('findMany', {
        uri: args.uri,
        pipeline: query,
        collection: args.collection,
        database: args.database,
    })
        .then(function (result) {
        return result;
    });
}
exports.findMany = findMany;
