// cypress-mongodb/dist/commands/insert.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.insertMany = exports.insertOne = void 0;
function insertOne(document, collection, database) {
    var args = {
        uri: Cypress.env('MONGODB_URI'),
        database: Cypress.env('MONGODB_DATABASE'),
        collection: Cypress.env('MONGODB_COLLECTION'),
        pipeline: [],
    };
    if (!document) {
        throw new Error('Document must be specified');
    }
    if (collection) {
        args.collection = collection;
    }
    else if (!args.collection) {
        throw new Error('Collection not specified');
    }
    if (database) {
        args.database = database;
    }
    else if (!args.database) {
        throw new Error('Database not specified');
    }
    return cy
        .task('insertOne', {
        uri: args.uri,
        pipeline: document,
        collection: args.collection,
        database: args.database,
    })
        .then(function (result) {
        return result;
    });
}
exports.insertOne = insertOne;
function insertMany(documents, collection, database) {
    var args = {
        uri: Cypress.env('MONGODB_URI'),
        database: Cypress.env('MONGODB_DATABASE'),
        collection: Cypress.env('MONGODB_COLLECTION'),
        pipeline: [],
    };
    if (!documents) {
        throw new Error('Documents must be specified');
    }
    if (collection) {
        args.collection = collection;
    }
    else if (!args.collection) {
        throw new Error('Collection not specified');
    }
    if (database) {
        args.database = database;
    }
    else if (!args.database) {
        throw new Error('Database not specified');
    }
    return cy
        .task('insertMany', {
        uri: args.uri,
        pipeline: documents,
        collection: args.collection,
        database: args.database,
    })
        .then(function (result) {
        return result;
    });
}
exports.insertMany = insertMany;
